internal/controllers/clusterextension_controller_test.go

```diff
@@ -1400,3 +1411,122 @@ func TestSetDeprecationStatus(t *testing.T) {
 		})
 	}
 }
+
+type MockActionGetter struct {
+	rels []*release.Release
+	err  error
+}
+
+func (mag *MockActionGetter) ActionClientFor(ctx context.Context, obj client.Object) (helmclient.ActionInterface, error) {
+	return mag, nil
+}
+
+func (mag *MockActionGetter) Get(name string, opts ...helmclient.GetOption) (*release.Release, error) {
+	return nil, nil
+}
+
+// This is the function we are really testing
+func (mag *MockActionGetter) History(name string, opts ...helmclient.HistoryOption) ([]*release.Release, error) {
+	return mag.rels, mag.err
+}
+
+func (mag *MockActionGetter) Install(name, namespace string, chrt *chart.Chart, vals map[string]interface{}, opts ...helmclient.InstallOption) (*release.Release, error) {
+	return nil, nil
+}
+
+func (mag *MockActionGetter) Upgrade(name, namespace string, chrt *chart.Chart, vals map[string]interface{}, opts ...helmclient.UpgradeOption) (*release.Release, error) {
+	return nil, nil
+}
+
+func (mag *MockActionGetter) Uninstall(name string, opts ...helmclient.UninstallOption) (*release.UninstallReleaseResponse, error) {
+	return nil, nil
+}
+
+func (mag *MockActionGetter) Reconcile(rel *release.Release) error {
+	return nil
+}
+
+func TestGetInstalledBundleHistory(t *testing.T) {
```

Comment:
Nit: This seems ripe for a table-driven test sort of thing where we declare a test case struct with the mock history/error, and expected return values. Then a loop over the test cases where we setup the mock, call the getter, and then make the assertion.